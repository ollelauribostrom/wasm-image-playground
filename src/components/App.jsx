import React from 'react';
import { Provider } from '../store/AppStore';
import ErrorBoundary from './ErrorBoundary';
import Reactor from './Reactor';
import Count from './Count';
import '../styles/main.css';

export default function App() {
  return (
    <Provider>
      <ErrorBoundary>
        <Reactor />
        <Count />
      </ErrorBoundary>
    </Provider>
  );
}
