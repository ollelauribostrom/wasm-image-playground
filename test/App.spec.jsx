import React from 'react';
import { mount, shallow } from 'enzyme';
import App from '../src/components/App';

function wait(duration) {
  return new Promise(resolve => {
    setTimeout(resolve, duration);
  });
}

jest.mock('../src/services/ColorService', () => ({
  getRandomColor: jest.fn(() => Promise.resolve(`hsla(${Math.random() * 360}, 100%, 70%, 1)`))
}));

describe('Tests for App', () => {
  test('it renders correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
  test('it handles clicks', async () => {
    const wrapper = mount(<App />);
    expect(wrapper.find('.reactor-count')).toHaveText('CLICK THE REACTOR');
    wrapper.find('.reactor-icon').simulate('click');
    await wait(10);
    expect(wrapper.find('.reactor-count')).toHaveText('CLICKED 1 TIME');
    wrapper.find('.reactor-icon').simulate('click');
    await wait(10);
    expect(wrapper.find('.reactor-count')).toHaveText('CLICKED 2 TIMES');
    wrapper.unmount();
  });
});
