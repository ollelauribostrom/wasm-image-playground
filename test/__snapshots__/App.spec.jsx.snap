// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for App it renders correctly 1`] = `
<o>
  <ErrorBoundary>
    <Connect(Reactor) />
    <Connect(Count) />
  </ErrorBoundary>
</o>
`;
